modal/stub.py

```diff
@@ -870,32 +892,41 @@ def cls(
         cloud: Optional[str] = None,  # Cloud provider to run the function on. Possible values are aws, gcp, auto.
     ) -> type:
         function_handles: Dict[str, _FunctionHandle] = {}
+        partial_functions: Dict[str, _PartialFunction] = {}
+
+        if hasattr(user_cls, "remote"):
+            raise InvalidError(f"Class {user_cls} has an attribute named 'remote' [TODO: expand]")
+
+        kwargs = dict(
```

Comment:
instead of creating kwargs here, i think you can just call `self.function(... all the arguments ...)` and then get back a Callable[[Callable], Callable]` or something like that which you can use to create the function handles and also pass it to `_make_remote_cls`

just marginally nicer since you get some sort of type safety

Reply:
nbd if i'm being unclear – i can change this later